lite/examples/sound_classification_use_task_api/ios/SoundClassification/TFlite/SoundClassificationHelper.swift

```diff
@@ -0,0 +1,112 @@
+  private var classifier: AudioClassifier
+  private var timer: Timer?
+  private let processQueue = DispatchQueue(label: "processQueue")
+
+  // MARK: - Initialization
+
+  /// A failable initializer for `SoundClassificationHelper`. A new instance is created if the model and
+  /// labels files are successfully loaded from the app's main bundle.
+  init?(modelType: ModelType, threadCount: Int, scoreThreshold: Float, maxResults: Int) {
+    let modelFilename = modelType.fileName
+
+    // Construct the path to the model file.
+    guard let modelPath = Bundle.main.path(
+      forResource: modelFilename,
+      ofType: "tflite"
+    ) else {
+      print("Failed to load the model file with name: \(modelFilename).")
+      return nil
+    }
+
+    // Specify the options for the `Classifier`.
+    let classifierOptions = AudioClassifierOptions(modelPath: modelPath)
+    classifierOptions.baseOptions.computeSettings.cpuSettings.numThreads = threadCount
+    classifierOptions.classificationOptions.maxResults = maxResults
+    classifierOptions.classificationOptions.scoreThreshold = scoreThreshold
+    do {
+      // Create the `Classifier`.
+      classifier = try AudioClassifier.classifier(options: classifierOptions)
+    } catch let error {
+      print("Failed to create the interpreter with error: \(error.localizedDescription)")
+      return nil
+    }
+  }
+
+  /// This class handles all data preprocessing and delegate results to Controller when classtifi is done
+  func runClassifier(overLap: Double) {
+    let inputAudioTensor = classifier.createInputAudioTensor()
+    let audioFormat = inputAudioTensor.audioFormat
+    let audioTensor = AudioTensor(audioFormat: audioFormat, sampleCount: inputAudioTensor.bufferSize)
+    do {
+      let audioRecord = try classifier.createAudioRecord()
+      func processing() {
+        let startTime = Date().timeIntervalSince1970
+        do {
+          try audioTensor.load(audioRecord: audioRecord)
+          let results = try classifier.classify(audioTensor: audioTensor)
+          let inferenceTime = Date().timeIntervalSince1970 - startTime
+          DispatchQueue.main.async {
+            // Send datas to Controller
+            self.delegate?.sendResult(
+              Result(inferenceTime: inferenceTime,
+                     categories: results.classifications[0].categories)
+            )
+          }
+        } catch {
+          print(error.localizedDescription)
+        }
+      }
+      // Start recording audio
+      try audioRecord.startRecording()
+      // Calculate interval between sampling based on overlap
+      let lengthInMilliSeconds = Double(inputAudioTensor.bufferSize) / Double(audioFormat.sampleRate)
+      let interval = lengthInMilliSeconds * Double(1 - overLap)
+      timer?.invalidate()
+      // Run the process after every fixed interval
+      timer = Timer.scheduledTimer(withTimeInterval: interval, repeats: true, block: { _ in
+        self.processQueue.async {
```

Comment:
You're creating circular references here. Timer will retain a strong reference to self, and self also has a strong reference to timer. The result is that the helper class and all of its variables will not be freed from memory when it's no longer needed.

You must weakify self here to avoid circular reference.